web/src/audio.rs

```diff
@@ -250,7 +251,8 @@ impl Buffer {
 impl Drop for Buffer {
     fn drop(&mut self) {
         if let Some(audio_node) = self.audio_node.take() {
-            audio_node.set_onended(None);
+            let scheduled: &AudioScheduledSourceNode = &audio_node;
```

Comment:
Is this a "JS downcast" of some sort? Or required by the borrow checker? 🤔 Sorry, not at PC right now.

Reply:
`AudioBufferSourceNode`'s `set_onended` method was deprecated in https://github.com/rustwasm/wasm-bindgen/pull/4020. This is a way of accessing the same method explicitly in its parent class.

Reply:
I see. So more of an upcast then.